tests/mxnet/test_shared_mem_store.py

```diff
@@ -111,47 +112,46 @@ def _check_compute_func(worker_id, graph_name):
         except:
             print("fail to connect to the graph store server.")
             time.sleep(1)
-    g._sync_barrier()
-    in_feats = g.nodes[0].data['feat'].shape[1]
-
-    # Test update all.
-    g.update_all(fn.copy_src(src='feat', out='m'), fn.sum(msg='m', out='preprocess'))
-    adj = g.adjacency_matrix()
-    tmp = mx.nd.dot(adj, g.nodes[:].data['feat'])
-    assert np.all((g.nodes[:].data['preprocess'] == tmp).asnumpy())
-    g._sync_barrier()
-    check_array_shared_memory(g, worker_id, [g.nodes[:].data['preprocess']])
-
-    # Test apply nodes.
-    data = g.nodes[:].data['feat']
-    g.apply_nodes(func=lambda nodes: {'feat': mx.nd.ones((1, in_feats)) * 10}, v=0)
-    assert np.all(data[0].asnumpy() == g.nodes[0].data['feat'].asnumpy())
-
-    # Test apply edges.
-    data = g.edges[:].data['feat']
-    g.apply_edges(func=lambda edges: {'feat': mx.nd.ones((1, in_feats)) * 10}, edges=0)
-    assert np.all(data[0].asnumpy() == g.edges[0].data['feat'].asnumpy())
-
-    g.init_ndata('tmp', (g.number_of_nodes(), 10), 'float32')
-    data = g.nodes[:].data['tmp']
-    # Test pull
-    g.pull(1, fn.copy_src(src='feat', out='m'), fn.sum(msg='m', out='tmp'))
-    assert np.all(data[1].asnumpy() == g.nodes[1].data['preprocess'].asnumpy())
-
-    # Test send_and_recv
-    in_edges = g.in_edges(v=2)
-    g.send_and_recv(in_edges, fn.copy_src(src='feat', out='m'), fn.sum(msg='m', out='tmp'))
-    assert np.all(data[2].asnumpy() == g.nodes[2].data['preprocess'].asnumpy())
-
-    g.destroy()
 
-def check_compute_func(worker_id, graph_name, return_dict):
     try:
-        _check_compute_func(worker_id, graph_name)
+        g._sync_barrier()
```

Comment:
Question: if some worker fails during the test and cannot reach this sync_barrier, what will happen?

Reply:
I added a timeout in _sync_barrier. I also added a test to check if the timeout works correctly.

Reply:
Cool. Also, the new CI config has timeout protection of all the unittest/example/tutorial tests, so we are safe now :).